### PR TITLE
API contract for updating user roles

### PR DESCRIPTION
**Pull request by vinayak-trivedi**
RFC: 
- https://github.com/Real-Dev-Squad/website-backend/issues/1093

API contract for API for updating user roles

### CODE REVIEW

File: users/README.md [hunk 1]
```diff
@@ -265,3 +266,36 @@ Updates data of the User.
...
+  None
+- **Query**  
+  None
+- **Headers**  
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Body**
+  `{
+    rolename: <boolean>
```
Comment (RitikJaiswal75) [inline users/README.md:284]: please provide allowed roles here

Comment (vikhyat187) [reply users/README.md:284]: +1

File: users/README.md [hunk 2]
```diff
@@ -265,3 +266,36 @@ Updates data of the User.
   - **Code:** 503
     - **Content:**
       `{ 'statusCode': 503, 'error': 'Service Unavailable', 'message': 'Something went wrong please contact admin' }`
+
+## **PATCH /users/:id/roles**
+
+Updates roles for the User.
+
+- **Params**  
+  None
```
Comment (vikhyat187) [inline users/README.md:275]: We do have `Id` in the params right?

File: users/README.md [hunk 3]
```diff
@@ -43,6 +43,7 @@ number and email address.
 |   [GET /users/:userId/badges](#get-usersidbadges)   | Returns badges assigned to the user  |
 |             [POST /users](#post-users)              |          Creates a new User          |
 |        [PATCH /users/self](#patch-usersself)        |       Updates data of the User       |
+|        [PATCH /users/:id/roles](#patch-usersidroles)        |       Updates user roles     |
```
Comment (heyrandhir) [inline users/README.md:46]: the formatting seems to be out of place 🤔

File: users/README.md (hunk 1 above)
Comment (heyrandhir) [reply users/README.md:284]: +2

File: users/README.md [hunk 4]
```diff
@@ -265,3 +266,36 @@ Updates data of the User.
...
+      `{ 'statusCode': 404, 'error': 'Not Found', 'message': 'User not found' }`
+  - **Code:** 400
+    - **Content:**
+      `{ 'statusCode': 400, 'error': 'Invalid Request', 'message': 'Invalid role' }`
+  - **Code:** 401
+    - **Content:**
+      `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 503
+    - **Content:**
+      `{ 'statusCode': 503, 'error': 'Service Unavailable', 'message': 'Something went wrong please contact admin' }`
```
Comment (heyrandhir) [inline users/README.md:301]: we should have 409 as well if the role trying to be updated already pre-exist.

File: users/README.md [hunk 5]
```diff
@@ -265,3 +266,36 @@ Updates data of the User.
...
+  Content-Type: application/json
+- **Cookie**  
+  rds-session: `<JWT>`
+- **Body**
+  `{
+    rolename: <boolean>
+  }`
+- **Success Response:**
+  - **Code:** 204
+    - **Content:** `{ 'message': 'role updated successfully!'}`
```
Comment (heyrandhir) [inline users/README.md:288]: Along with message we should also return the updated data. Returning the updated data can be useful for clients that need to display the updated information to the user.

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: users/README.md (hunk 1 above)
Comment (shubhamsigdar1) [reply users/README.md]: Done can you check in latest commit

File: users/README.md (hunk 2 above)
Comment (shubhamsigdar1) [reply users/README.md]: Done can you check in latest commit

File: users/README.md (hunk 3 above)
Comment (shubhamsigdar1) [reply users/README.md]: Done can you please check in latest commit

File: users/README.md (hunk 4 above)
Comment (shubhamsigdar1) [reply users/README.md]: Done can you please check in latest commit

File: users/README.md (hunk 5 above)
Comment (shubhamsigdar1) [reply users/README.md]: Its wrong success status will be 200

File: users/README.md [hunk 6]
```diff
@@ -265,3 +266,40 @@ Updates data of the User.
...
+- **Body**
+  `{
+  member: <boolean>
+  archived: <boolean>
+}`
+- **Success Response:**
+  - **Code:** 200
+    - **Content:** `{ 'message': 'role updated successfully!'}`
+- **Error Response:**
+- **Code:** 409
```
Comment (heyrandhir) [inline users/README.md:291]: this code is not at same indentation as others

File: users/README.md [hunk 7]
```diff
@@ -265,3 +266,40 @@ Updates data of the User.
...
+    - **Content:**
+      `{ 'statusCode': 404, 'error': 'Not Found', 'message': 'User not found' }`
+  - **Code:** 400
+    - **Content:**
+      `{ 'statusCode': 400, 'error': 'Invalid Request', 'message': 'Invalid role' }`
+  - **Code:** 401
+    - **Content:**
+      `{ 'statusCode': 401, 'error': 'Unauthorized', 'message': 'Unauthenticated User' }`
+  - **Code:** 503
+    - **Content:**
```
Comment (heyrandhir) [inline users/README.md:304]: it should be 500 if its an internal server error. are you sure its 503 ?

**Review by heyrandhir — CHANGES_REQUESTED** [review]

File: users/README.md (hunk 6 above)
Comment (heyrandhir) [reply users/README.md:291]: should be 1 level deep like others

File: users/README.md (hunk 5 above)
Comment (heyrandhir) [reply users/README.md]: Success status 200 is fine. what I meant was along with the response you could also return the updated role values. Its optional but provide more value to the one consuming this API

**Review by heyrandhir — DISMISSED** [review]

**Review by DashDeipayan — APPROVED** [review]
LGTM

**Pull request merged**